### DISCUSSION TITLE
ContentControl parent binding

### DISCUSSION THREAD

**Discussion by montex12** [AvaloniaUI/Avalonia#12272] [Q&A]
Hello,

I have a `ContentControl` hosted in an `UserControl`:
```XML
<UserControl x:Class="MainPage"
    .....
  <ContentControl Content="{Binding SubPage}" />
</UserControl>
```
That `ContentControl` have some buttons that I would like to bind in the parent `UserControl`.
```XML
<UserControl x:Class="SubPage1"
    .....
  <StackPanel>
    <Button Command="{Binding $parent.MainPageCommand1}" />
    <Button Command="{Binding $parent.MainPageCommand2}" />
  </StackPanel>
</UserControl>
```

`MainPageCommand1` and `MainPageCommand2` are defined in the MainPage ViewModel class.
```C#
public class MainPageViewModel : ReactiveObject
{
    public ICommand MainPageCommand1 { get; }
    public ICommand MainPageCommand2 { get; }
}
```

Is there a way?

**Comment by timunie** [comment]
$parent is the logical parent control. however, you want to access it's DataContext. So it becomes 
`$parent.((vm:ParentVM)DataContext).MyCommand`. 

Note: the cast is important to make compiled binidngs work. 

more: 
https://docs.avaloniaui.net/docs/next/guides/data-binding/binding-to-controls#binding-to-an-ancestor
https://docs.avaloniaui.net/docs/next/basics/data/data-binding/compiled-bindings
